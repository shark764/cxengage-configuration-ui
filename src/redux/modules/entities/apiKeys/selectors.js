import { Map } from 'immutable';
import { getSelectedEntity } from '../selectors';
import { selectTenantRoles } from '../roles/selectors';

export const getInitialUpdateFormValues = state =>
  new Map({
    name: getSelectedEntity(state).get('name'),
    description: getSelectedEntity(state).get('description'),
    roleId: getSelectedEntity(state).get('roleId'),
    id: getSelectedEntity(state).get('id'),
    secret: getSelectedEntity(state).get('secret')
  });

export const getInitialCreateFormValues = state =>
  new Map({
    roleId: selectTenantRoles(state).length > 0 ? selectTenantRoles(state)[0].value : ''
  });
