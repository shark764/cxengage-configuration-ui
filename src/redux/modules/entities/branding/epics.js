/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import 'rxjs/add/operator/mergeMap';
import { fetchData } from '../index';

export const FetchBranding = action$ =>
  action$
    .ofType('FETCH_BRANDING_$')
    .mergeMap(() => [fetchData('branding'), fetchData('protectedBranding')]);
