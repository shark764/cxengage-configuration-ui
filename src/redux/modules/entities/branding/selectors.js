/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

export const getBranding = state =>
  state.getIn(['Entities', 'branding', 'data']);

export const getProtectedBranding = state =>
  state.getIn(['Entities', 'protectedBranding', 'data']);
