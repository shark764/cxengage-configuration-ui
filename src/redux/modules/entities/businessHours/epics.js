import 'rxjs/add/operator/map';
import 'rxjs/add/operator/filter';

import { fromPromise } from 'rxjs/observable/fromPromise';
import { change } from 'redux-form';
import { removeLastLetter, camelCaseToRegularFormAndRemoveLastLetter } from 'serenova-js-utils/strings';

import { sdkPromise } from '../../../../utils/sdk';
import { handleSuccess, handleError } from '../handleResult';
import { entitiesMetaData } from '../metaData';

const days = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];

export const changeBusinessHoursType = (action$, store) =>
  action$
    .ofType('@@redux-form/CHANGE')
    .filter(a => a.meta.form.includes('businessHours') && a.meta.field === 'businessHoursType')
    .mergeMap(a => {
      return days
        .reduce((r, i) => [...r, i.concat('StartTimeMinutes'), i.concat('EndTimeMinutes')], [])
        .map(i => change(a.meta.form, i, a.payload === 'scheduledHours' && i.includes('EndTimeMinutes') ? 1440 : 0));
    });

export const CreateBusinessHours = action$ =>
  action$
    .ofType('CREATE_ENTITY')
    .filter(({ entityName }) => entityName === 'businessHours')
    .map(a => {
      a.sdkCall = entitiesMetaData['businessHours'].entityApiRequest('create', 'singleMainEntity');
      const timeMinutes = Object.entries(a.values)
        .filter(([key, value]) => key.includes('StartTimeMinutes') || key.includes('EndTimeMinutes'))
        .reduce(
          (r, [key, value]) => ({
            ...r,
            [key]:
              key.includes('EndTimeMinutes') && a.values.businessHoursType === 'scheduledHours' && value === 0
                ? 1440
                : value
          }),
          {}
        );
      const { timezone, name, description } = a.values;

      a.sdkCall.data.timeMinutes = timeMinutes;
      a.sdkCall.data.timezone = timezone;
      a.sdkCall.data.active = true;
      a.sdkCall.data.name = name;
      a.sdkCall.data.description = description;

      return { ...a };
    })
    .concatMap(a =>
      fromPromise(sdkPromise(a.sdkCall))
        .map(response =>
          handleSuccess(
            response,
            a,
            `${camelCaseToRegularFormAndRemoveLastLetter('businessHours')} was created successfully!`
          )
        )
        .catch(error => handleError(error, a))
    );

export const UpdateBusinessHours = action$ =>
  action$
    .ofType('UPDATE_ENTITY')
    .filter(({ entityName }) => entityName === 'businessHours')
    .map(a => {
      a.sdkCall = entitiesMetaData['businessHours'].entityApiRequest('update', 'singleMainEntity');
      const timeMinutes = Object.entries(a.values)
        .filter(([key, value]) => key.includes('StartTimeMinutes') || key.includes('EndTimeMinutes'))
        .reduce(
          (r, [key, value]) => ({
            ...r,
            [key]:
              key.includes('EndTimeMinutes') && a.values.businessHoursType === 'scheduledHours' && value === 0
                ? 1440
                : value
          }),
          {}
        );
      const { timezone, name, description } = a.values;

      a.sdkCall.data.timeMinutes = timeMinutes;
      a.sdkCall.data.timezone = timezone;
      a.sdkCall.data.name = name;
      a.sdkCall.data.description = description;
      a.sdkCall.data[removeLastLetter('businessHours') + 'Id'] = a.entityId;

      return { ...a };
    })
    .concatMap(a =>
      fromPromise(sdkPromise(a.sdkCall))
        .map(response =>
          handleSuccess(
            response,
            a,
            `${camelCaseToRegularFormAndRemoveLastLetter('businessHours')} was updated successfully!`
          )
        )
        .catch(error => handleError(error, a))
    );
