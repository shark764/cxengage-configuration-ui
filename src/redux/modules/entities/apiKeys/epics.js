/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import 'rxjs/add/operator/map';
import 'rxjs/add/operator/filter';
import { fromPromise } from 'rxjs/observable/fromPromise';
import { entitiesMetaData } from '../metaData';
import { camelCaseToRegularFormAndRemoveLastLetter } from 'serenova-js-utils/strings';
import { getSelectedEntityId, getCurrentEntity } from '../selectors';

import { sdkPromise } from '../../../../utils/sdk';
import { handleSuccess, handleError } from '../handleResult';

export const getRolesAfterFetchingApiKeys = action$ =>
  action$
    .ofType('FETCH_DATA_FULFILLED')
    .filter(a => a.entityName === 'apiKeys')
    .map(a => ({ type: 'FETCH_DATA', entityName: 'roles' }));

export const UpdateApiKey = action$ =>
  action$
    .ofType('UPDATE_ENTITY')
    .filter(({ entityName }) => entityName === 'apiKeys')
    .map(a => {
      a.sdkCall = entitiesMetaData['apiKeys'].entityApiRequest('update', 'singleMainEntity');

      const { name, description, roleId, status } = a.values;

      a.sdkCall.data = { status, roleId, name, description, apiKeyId: a.entityId };
      return { ...a };
    })
    .concatMap(a =>
      fromPromise(sdkPromise(a.sdkCall))
        .map(response =>
          handleSuccess(
            response,
            a,
            `${camelCaseToRegularFormAndRemoveLastLetter('apiKeys')} was updated successfully!`
          )
        )
        .catch(error => handleError(error, a))
    );

export const DeteleApiKey = (action$, store) =>
  action$
    .ofType('DELETE_API_KEY')
    .map(a => ({
      ...a,
      entityName: getCurrentEntity(store.getState())
    }))
    .filter(({ entityName }) => entityName === 'apiKeys')
    .map(a => {
      a.sdkCall = {
        command: 'deleteApiKey',
        data: { apiKeyId: getSelectedEntityId(store.getState()) },
        module: 'entities',
        topic: 'cxengage/entities/delete-api-key-response'
      };
      return { ...a };
    })
    .concatMap(a =>
      fromPromise(sdkPromise(a.sdkCall))
        .map(response =>
          handleSuccess(
            response,
            a,
            `${camelCaseToRegularFormAndRemoveLastLetter('apiKeys')} was deleted successfully!`
          )
        )
        .catch(error => handleError(error, a))
    );
