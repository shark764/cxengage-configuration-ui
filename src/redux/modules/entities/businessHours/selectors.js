import { selectFormInitialValues } from '../../form/selectors';
import { getSelectedEntity } from '../selectors';
import { Map } from 'immutable';

export const selectBusinessHoursFormInitialValues = state => {
  if (getSelectedEntity(state) === undefined) {
    const initialTimes = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'].reduce(
      (r, key) => ({ ...r, [key.concat('StartTimeMinutes')]: 0, [key.concat('EndTimeMinutes')]: 0 }),
      {}
    );
    return new Map({ active: true, businessHoursType: '24/7', timezone: 'Etc/GMT+12', ...initialTimes });
  }
  return selectFormInitialValues(state);
};
