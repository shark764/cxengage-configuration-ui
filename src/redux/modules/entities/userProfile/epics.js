import 'rxjs/add/operator/map';
import 'rxjs/add/operator/filter';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/switchMap';
import { fromPromise } from 'rxjs/observable/fromPromise';
import { of } from 'rxjs/observable/of';
import { initialize } from 'redux-form';
import { fromJS } from 'immutable';
import { Toast } from 'cx-ui-components';

import { sdkPromise, errorManager } from '../../../../utils/sdk';
import { fetchData } from '../index';
import { getCurrentAgentId } from '../../userData/selectors';
import { userProfileInitialValues } from './selectors';
import { handleError, handleSuccess } from '../handleResult';
import { entitiesMetaData } from '../metaData';

export const FetchUserProfileData = (action$, store) =>
  action$
    .ofType('FETCH_DATA')
    .filter(({ entityName }) => entityName === 'userProfile')
    .switchMap(a =>
      fromPromise(
        sdkPromise({
          command: 'getTenantDetails',
          module: 'session',
          topic: 'cxengage/session/get-tenant-details'
        })
      )
        .map(response => ({ ...a, response }))
        .catch(error => ({ ...a, error }))
    )
    .switchMap(
      ({ response, error, ...a }) =>
        error
          ? handleError(error, a)
          : [
              handleSuccess(response, { ...a, type: 'USER_PROFILE_TENANTS' }),
              fetchData('integrations'),
              {
                type: 'FETCH_DATA_ITEM',
                entityName: 'users',
                id: getCurrentAgentId(store.getState())
              }
            ]
    );

// Reinitialize the forms when we sucesfully updated the userProfile or when the requests to fetch info for user profile
// are completed succesfully
export const ReInitializeUserProfileForms = (action$, store) =>
  action$
    .ofType(
      'UPDATE_USER_PROFILE_FULFILLED',
      'USER_PROFILE_TENANTS_FULFILLED',
      'FETCH_DATA_ITEM_FULFILLED',
      'FETCH_DATA_FULFILLED'
    )
    .filter(
      ({ type, id, entityName }) =>
        (type === 'FETCH_DATA_FULFILLED' && entityName === 'integrations') ||
        (type === 'FETCH_DATA_ITEM_FULFILLED' &&
          entityName === 'users' &&
          id === getCurrentAgentId(store.getState())) ||
        (type !== 'FETCH_DATA_FULFILLED' && type !== 'FETCH_DATA_ITEM_FULFILLED')
    )
    .switchMap(() => {
      const initialValues = userProfileInitialValues(store.getState());
      return [
        initialize('userProfile:details', initialValues.delete('extensions'), false),
        initialize('userProfile:extensions', fromJS({ extensions: initialValues.get('extensions') }), false)
      ];
    });

export const UpdateUserProfile = (action$, store) =>
  action$
    .ofType('UPDATE_USER_PROFILE')
    .switchMap(
      ({
        values: {
          extensions,
          password,
          currentPassword,
          firstName,
          lastName,
          defaultTenant,
          workStationId,
          noPassword,
          defaultIdentityProvider,
          externalId,
          username
        },
        profileEntity,
        ...a
      }) => {
        const userId = getCurrentAgentId(store.getState());
        return fromPromise(
          sdkPromise(
            profileEntity === 'platformUser'
              ? {
                  command: 'updatePlatformUser',
                  data: {
                    updateBody: {
                      firstName,
                      lastName,
                      defaultTenant,
                      ...(currentPassword &&
                        password && {
                          password,
                          currentPassword
                        }),
                      externalId
                    },
                    userId
                  },
                  module: 'entities',
                  topic: 'cxengage/entities/update-platform-user-response'
                }
              : {
                  ...entitiesMetaData['users'].entityApiRequest('update', 'singleMainEntity'),
                  data: {
                    updateBody: {
                      extensions: extensions.map(({ id, hide, ...ext }) =>
                        Object.entries(ext).reduce((obj, [key, value]) => (value ? { ...obj, [key]: value } : obj), {})
                      ),
                      defaultIdentityProvider,
                      noPassword,
                      workStationId
                    },
                    userId
                  }
                }
          )
        )
          .map(response => ({ ...a, currentPassword, password, profileEntity, response, userId, username }))
          .catch(error => of({ error, ...a }));
      }
    )
    .switchMap(({ currentPassword, password, profileEntity, username, userId, error, ...a }) => {
      const successfulResponse = {
        ...a,
        userId
      };

      if (profileEntity === 'platformUser' && currentPassword && password && !error) {
        return fromPromise(
          sdkPromise({
            command: 'login',
            module: 'authentication',
            data: {
              username,
              password,
              noStateReset: true
            },
            topic: 'cxengage/sessions/get-new-token-response'
          })
        )
          .map(() => successfulResponse)
          .catch(error => of({ error, ...a }));
      } else if (error) {
        return of({ error, ...a });
      } else {
        return of(successfulResponse);
      }
    })
    .map(({ error, response, ...a }) => {
      if (error) {
        Toast.error(errorManager(error).errorMessage);
        return {
          ...a,
          type: `${a.type}_REJECTED`
        };
      } else {
        return handleSuccess(response, a, 'Profile updated succesfully');
      }
    });
