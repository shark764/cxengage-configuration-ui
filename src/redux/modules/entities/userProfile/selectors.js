import { createSelector } from 'reselect';
import { generateUUID } from 'serenova-js-utils/uuid';
import { fromJS, Map } from 'immutable';

import { findEntity } from '../selectors';
import { getCurrentAgentId } from '../../userData/selectors';
import { getUpdatedUserExtensions } from '../users/selectors';

const userTenants = (state) => state.getIn(['Entities', 'userProfile', 'userTenants']);

export const selectUserTenants = createSelector(
  userTenants,
  (tenants) =>
    tenants &&
    tenants.toJS().map(({ tenantId, name }) => ({
      value: tenantId,
      label: name,
    }))
);

export const isSaving = (state) => state.getIn(['Entities', 'userProfile', 'updating']);

export const userProfileInitialValues = (state) => {
  const userData = findEntity(state, 'users', getCurrentAgentId(state));
  return userData
    ? userData.update('extensions', (extensions) =>
        fromJS(
          getUpdatedUserExtensions(state, extensions.toJS().map((extension) => ({ ...extension, id: generateUUID() })))
        )
      )
    : Map({});
};
