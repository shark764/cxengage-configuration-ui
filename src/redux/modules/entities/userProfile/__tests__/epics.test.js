import { fromJS } from 'immutable';
import { ActionsObservable } from 'redux-observable';

import { sdkPromise, errorManager } from '../../../../../utils/sdk';
import { getCurrentAgentId } from '../../../userData/selectors';
import { userProfileInitialValues } from '../selectors';

import { FetchUserProfileData, ReInitializeUserProfileForms, UpdateUserProfile } from '../epics';

import { mockStore } from '../../../../../utils/testUtils';

jest.mock('../../../../../utils/sdk');
jest.mock('../../../userData/selectors');
jest.mock('../selectors');

getCurrentAgentId.mockReturnValue('user-id');
userProfileInitialValues.mockReturnValue(fromJS({
    email: 'email@email.com',
    firstName: 'Whatever',
    lastName: 'Whoever',
    defaultTenant: 'tenant-id',
    extensions: [
        {
            type: 'WebRTC',
            name: 'whatever',
            id: 'an-id',
            hide: false
        },
        {
            type: 'PSTN',
            name: 'someone',
            id: 'another-id',
            hide: false
        }
    ],
    skills: [
        {
            name: 'skill 1'
        },
        {
            name: 'skill 2'
        }
    ],
    groups: [
        {
            name: 'group 1'
        },
        {
            name: 'group 2'
        }
    ],
    externalId: 'external-id',
    defaultIdentityProvider: 'identityProvider',
    noPassword: false,
    workStationId: 'workstation-id'
}));

describe('FetchUserProfileData', () => {
    beforeAll(() => {
        sdkPromise
            .mockResolvedValueOnce({
                details: [{
                        value: 'tenant1-id',
                        label: 'tenant 1'
                    },
                    {
                        value: 'tenant2-id',
                        label: 'tenant 2'
                    }
                ]
            })
            .mockRejectedValue(new Error('error'))
    });

    it('Fetches tenants the user can be assigned to, and emits actions to fetch integrations data and user info', done => {
        FetchUserProfileData(ActionsObservable.of({
            type: 'FETCH_DATA',
            entityName: 'userProfile'
        }), mockStore).subscribe(actualOutputActions => {
            expect(actualOutputActions).toMatchSnapshot();
            done();
        });
    });

    it('API errors and action to handle the error is emitted', done => {
        FetchUserProfileData(ActionsObservable.of({
            type: 'FETCH_DATA',
            entityName: 'userProfile'
        }), mockStore).subscribe(actualOutputActions => {
            expect(actualOutputActions).toMatchSnapshot();
            done();
        });
    });

    afterAll(() => {
        sdkPromise.mockReset();
    })
});

describe('ReInitializeUserProfileForms', () => {
    it('Reinitialize the forms when we sucesfully updated the userProfile or when the requests to fetch info for user profile are completed succesfully', done => {
        ReInitializeUserProfileForms(ActionsObservable.of({
            type: 'FETCH_DATA_ITEM_FULFILLED',
            entityName: 'users',
            id: 'user-id'
        }), mockStore).subscribe(actualOutputActions => {
            expect(actualOutputActions).toMatchSnapshot();
            done();
        });
    })
});

describe('UpdateUserProfile', () => {
    const values = {    
        email: 'email@email.com',
        firstName: 'Whatever',
        lastName: 'Whoever',
        defaultTenant: 'tenant-id',
        extensions: [
            {
                type: 'WebRTC',
                name: 'whatever',
                id: 'an-id',
                hide: false
            },
            {
                type: 'PSTN',
                name: 'someone',
                id: 'another-id',
                hide: false
            }
        ],
        externalId: 'external-id',
        defaultIdentityProvider: 'identityProvider',
        noPassword: false,
        workStationId: 'workstation-id',
        currentPassword: 'Password1!',
        password: 'Password2!'
    };

    describe('Current entity being update it\'s the platform user with a new password', () => {
        beforeAll(() => {
            sdkPromise.mockResolvedValue({
                result: 'mock value'
            });
        });

        it('All requests are made succesfully and UPDATE_USER_PROFILE_FULFILLED it\'s emitted', done => {
            UpdateUserProfile(ActionsObservable.of({
                type: 'UPDATE_USER_PROFILE',
                values,
                profileEntity: 'platformUser' 
            }), mockStore).subscribe(actualOutputActions => {
                expect(actualOutputActions).toMatchSnapshot();
                done();
            });
        });

        it('sdkPromise get\'s called twice', () => {
            expect(sdkPromise.mock.calls.length).toBe(2);
        })

        afterAll(() => {
            sdkPromise.mockReset();
        });
    });

    describe('Current entity being update it\'s the platform user but no new password', () => {
        beforeAll(() => {
            sdkPromise.mockResolvedValue({
                result: 'mock value'
            });
        });

        it('Request is to update user profile made succesfully and UPDATE_USER_PROFILE_FULFILLED it\'s emitted', done => {
            UpdateUserProfile(ActionsObservable.of({
                type: 'UPDATE_USER_PROFILE',
                values: {
                    ...values,
                    password: undefined,
                    currentPassword: undefined
                },
                profileEntity: 'platformUser' 
            }), mockStore).subscribe(actualOutputActions => {
                expect(actualOutputActions).toMatchSnapshot();
                done();
            });
        });

        it('sdkPromise get\'s called once', () => {
            expect(sdkPromise.mock.calls.length).toBe(1);
        });

        afterAll(() => {
            sdkPromise.mockReset();
        });
    });

    describe('Current entity being update it\'s the tenant user(updating extensions)', () => {
        beforeAll(() => {
            sdkPromise.mockResolvedValue({
                result: 'mock value'
            });
        });

        it('Request is to update user profile made succesfully and UPDATE_USER_PROFILE_FULFILLED it\'s emitted', done => {
            UpdateUserProfile(ActionsObservable.of({
                type: 'UPDATE_USER_PROFILE',
                values: {
                    ...values,
                    password: undefined,
                    currentPassword: undefined
                },
                profileEntity: 'tenantUser' 
            }), mockStore).subscribe(actualOutputActions => {
                expect(actualOutputActions).toMatchSnapshot();
                done();
            });
        });

        it('sdkPromise get\'s called once', () => {
            expect(sdkPromise.mock.calls.length).toBe(1);
        });

        afterAll(() => {
            sdkPromise.mockReset();
        });
    });

    describe('Current entity being update it\'s platform user with a new password', () => {
        beforeAll(() => {
            sdkPromise.mockResolvedValueOnce({
                result: 'mock value'
            }).mockRejectedValue(new Error('error'));

            errorManager.mockReturnValue({
                errorMessage: 'error message'
            });
        });

        it('Request is to update the platform user it\'s made sucessfully but the request to update the password fails and a UPDATE_USER_PROFILE_REJECTED is emitted', done => {
            UpdateUserProfile(ActionsObservable.of({
                type: 'UPDATE_USER_PROFILE',
                values,
                profileEntity: 'platformUser'
            }), mockStore).subscribe(actualOutputActions => {
                expect(actualOutputActions).toMatchSnapshot();
                done();
            });
        });

        it('sdkPromise get\'s called twice', () => {
            expect(sdkPromise.mock.calls.length).toBe(2);
        });

        afterAll(() => {
            sdkPromise.mockReset();
            errorManager.mockReset();
        });
    });
});