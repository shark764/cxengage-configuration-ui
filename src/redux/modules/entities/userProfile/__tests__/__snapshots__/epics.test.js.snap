// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchUserProfileData API errors and action to handle the error is emitted 1`] = `
Object {
  "entityName": "userProfile",
  "type": "FETCH_DATA_REJECTED",
}
`;

exports[`FetchUserProfileData Fetches tenants the user can be assigned to, and emits actions to fetch integrations data and user info 1`] = `
Object {
  "entityName": "userProfile",
  "response": Object {
    "details": Array [
      Object {
        "label": "tenant 1",
        "value": "tenant1-id",
      },
      Object {
        "label": "tenant 2",
        "value": "tenant2-id",
      },
    ],
  },
  "type": "USER_PROFILE_TENANTS_FULFILLED",
}
`;

exports[`FetchUserProfileData Fetches tenants the user can be assigned to, and emits actions to fetch integrations data and user info 2`] = `
Object {
  "entityName": "integrations",
  "tableType": undefined,
  "type": "FETCH_DATA",
}
`;

exports[`FetchUserProfileData Fetches tenants the user can be assigned to, and emits actions to fetch integrations data and user info 3`] = `
Object {
  "entityName": "users",
  "id": "user-id",
  "type": "FETCH_DATA_ITEM",
}
`;

exports[`ReInitializeUserProfileForms Reinitialize the forms when we sucesfully updated the userProfile or when the requests to fetch info for user profile are completed succesfully 1`] = `
Object {
  "meta": Object {
    "form": "userProfile:details",
    "keepDirty": false,
  },
  "payload": Immutable.Map {
    "lastName": "Whoever",
    "skills": Immutable.List [
      Immutable.Map {
        "name": "skill 1",
      },
      Immutable.Map {
        "name": "skill 2",
      },
    ],
    "externalId": "external-id",
    "defaultTenant": "tenant-id",
    "defaultIdentityProvider": "identityProvider",
    "groups": Immutable.List [
      Immutable.Map {
        "name": "group 1",
      },
      Immutable.Map {
        "name": "group 2",
      },
    ],
    "firstName": "Whatever",
    "noPassword": false,
    "email": "email@email.com",
    "workStationId": "workstation-id",
  },
  "type": "@@redux-form/INITIALIZE",
}
`;

exports[`ReInitializeUserProfileForms Reinitialize the forms when we sucesfully updated the userProfile or when the requests to fetch info for user profile are completed succesfully 2`] = `
Object {
  "meta": Object {
    "form": "userProfile:extensions",
    "keepDirty": false,
  },
  "payload": Immutable.Map {
    "extensions": Immutable.List [
      Immutable.Map {
        "type": "WebRTC",
        "name": "whatever",
        "id": "an-id",
        "hide": false,
      },
      Immutable.Map {
        "type": "PSTN",
        "name": "someone",
        "id": "another-id",
        "hide": false,
      },
    ],
  },
  "type": "@@redux-form/INITIALIZE",
}
`;

exports[`UpdateUserProfile Current entity being update it's platform user with a new password Request is to update the platform user it's made sucessfully but the request to update the password fails and a UPDATE_USER_PROFILE_REJECTED is emitted 1`] = `
Object {
  "type": "UPDATE_USER_PROFILE_REJECTED",
}
`;

exports[`UpdateUserProfile Current entity being update it's the platform user but no new password Request is to update user profile made succesfully and UPDATE_USER_PROFILE_FULFILLED it's emitted 1`] = `
Object {
  "response": Object {
    "result": "mock value",
  },
  "type": "UPDATE_USER_PROFILE_FULFILLED",
  "userId": "user-id",
}
`;

exports[`UpdateUserProfile Current entity being update it's the platform user with a new password All requests are made succesfully and UPDATE_USER_PROFILE_FULFILLED it's emitted 1`] = `
Object {
  "response": Object {
    "result": "mock value",
  },
  "type": "UPDATE_USER_PROFILE_FULFILLED",
  "userId": "user-id",
}
`;

exports[`UpdateUserProfile Current entity being update it's the tenant user(updating extensions) Request is to update user profile made succesfully and UPDATE_USER_PROFILE_FULFILLED it's emitted 1`] = `
Object {
  "response": Object {
    "result": "mock value",
  },
  "type": "UPDATE_USER_PROFILE_FULFILLED",
  "userId": "user-id",
}
`;
